File: java/1894. Find the Student that Will Replace the Chalk.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        int n = chalk.length;
        long total = 0;
        for (int i : chalk) total += (long)i;
        k %= total;
        for (int i = 0; i < n; i++) {
            if (chalk[i] > k) return i;
            k -= chalk[i];
        }
        return -1;
    }
}

File: go/1894. Find the Student that Will Replace the Chalk.go
func chalkReplacer(chalk []int, k int) int {
	total := 0
	for _, v := range chalk {
		total += v
	}
	k %= total
	for i := 0; i < len(chalk); i++ {
		if chalk[i] > k {
			return i
		}
		k -= chalk[i]
	}
	return -1
}